src/poetry/mixology/version_solver.py

```diff
@@ -374,10 +375,20 @@ def _get_min(dependency: Dependency) -> Tuple[bool, int]:
                 )
                 return dependency.complete_name
 
-            try:
-                version = packages[0]
-            except IndexError:
-                version = None
+            version = None
+            if dependency.name not in self._use_latest:
+                # prefer locked version of compatible (not exact same) dependency;
+                # required in order to not unnecessarily update dependencies with
+                # extras, e.g. "coverage" vs. "coverage[toml]"
+                locked = self._locked.get(dependency.name, None)
+                if locked is not None:
+                    for version in packages:
+                        if version.version <= locked.version:
+                            if version.version != locked.version:
+                                version = None
+                            break
+            with suppress(IndexError):
+                version = version or packages[0]
```

Comment:
I feel like this line is a bit clunky -- do you think this is cleaner than a `None` check?

Reply:
Probably, a None check is easier to understand. I will change it.